src/main/java/com/ericsson/ei/subscription/InformSubscriber.java

```diff
@@ -123,168 +131,223 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
-     * method POST.
+     * Prepares headers to be used when making a rest call with the method POST.
      *
-     * @param notificationMeta
-     *    A String containing a URL
-     * @param subscriptionJson
-     *     Used to extract the rest post body media type from
+     * @param notificationMeta A String containing a URL
+     * @param subscriptionJson Used to extract the rest post body media type from
      * @return headers
+     * @throws AuthenticationException
      */
-    private HttpHeaders prepareHeaders(String notificationMeta, JsonNode subscriptionJson) {
+    private HttpHeaders prepareHeaders(String notificationMeta, JsonNode subscriptionJson)
+            throws AuthenticationException {
         HttpHeaders headers = new HttpHeaders();
-        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType", subscriptionJson);
+        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType",
+                subscriptionJson);
         headers.setContentType(MediaType.valueOf(headerContentMediaType));
-        LOGGER.debug("Successfully added header: 'restPostBodyMediaType':'{}'", headerContentMediaType);
+        LOGGER.debug("Successfully added header: 'restPostBodyMediaType':'{}'",
+                headerContentMediaType);
+
         headers = addAuthenticationData(headers, notificationMeta, subscriptionJson);
 
         return headers;
     }
 
     /**
-     * This method adds the authentication details to the headers.
+     * Adds the authentication details to the headers.
      *
      * @param headers
-     * @param notificationMeta
-     *     A String containing a URL
-     * @param subscriptionJson
-     *     Used to extract the authentication type from
+     * @param notificationMeta A String containing a URL
+     * @param subscriptionJson Used to extract the authentication type from
      * @return headers
+     * @throws AuthenticationException
      */
-    private HttpHeaders addAuthenticationData(HttpHeaders headers, String notificationMeta, JsonNode subscriptionJson) {
+    private HttpHeaders addAuthenticationData(HttpHeaders headers, String notificationMeta,
+            JsonNode subscriptionJson) throws AuthenticationException {
         String authType = getSubscriptionField("authenticationType", subscriptionJson);
-        if (!authType.equals("BASIC_AUTH")) {
-            return headers;
-        }
-
         String username = getSubscriptionField("userName", subscriptionJson);
         String password = getSubscriptionField("password", subscriptionJson);
 
-        if (!username.equals("") && !password.equals("")) {
-            String encoding = Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
-            headers.add("Authorization", "Basic " + encoding);
-            LOGGER.debug("Successfully added header for 'Authorization'");
+        boolean authorizationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
```

Comment:
I haven't followed the conversation about this very closely, but what was said about using authentication vs. authorization? :)

Reply:
Done